subprojects/core/src/main/java/org/gradle/process/internal/JavaExecAction.java

```diff
@@ -22,4 +22,6 @@
 @NonExtensible
 public interface JavaExecAction extends JavaExecSpec {
     ExecResult execute();
+
+    void cancel();
```

Comment:
Could this just implement `Cancellable`?  Meaning "this is a thing that can be cancelled when the build is cancelled".